server/controllers/student.py

```diff
@@ -132,6 +132,18 @@ def code(name, submit, bid):
     return render_template('student/assignment/code.html',
         course=assign.course, assignment=assign, backup=backup, use_diff=use_diff,
         files_before=assign.files, files_after=backup.files())
+        
+@student.route('/download/<hashid:bid>/<file>')
```

Comment:
I think this means that course id's can't start with download. I'm fine with adding download to the protected URLs list. 

Alternatively, the route could be `/<assignment_name:name>/submissions/<hashid:bid>/download/<file>`